File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/AuthService.java
package com.skedgo.tripkit.booking;

import com.skedgo.tripkit.common.model.Region;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import io.reactivex.Observable;

/**
 * Extension of {@link AuthApi} that facilitates fetching providers for a region.
 */
public interface AuthService extends AuthApi {
    Observable<List<AuthProvider>> fetchProvidersByRegionAsync(
        @NonNull Region region,
        @Nullable String mode,
        boolean bsb
    );
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/AuthService.kt
package com.skedgo.tripkit.booking

import com.skedgo.tripkit.common.model.region.Region
import io.reactivex.Observable

/**
 * Extension of [AuthApi] that facilitates fetching providers for a region.
 */
interface AuthService : AuthApi {
    fun fetchProvidersByRegionAsync(
        region: Region,
        mode: String?,
        bsb: Boolean
    ): Observable<List<AuthProvider>>
}